akka-actor/src/main/scala/akka/io/InetAddressDnsResolver.scala

```diff
@@ -94,22 +111,66 @@ class InetAddressDnsResolver(cache: SimpleDnsCache, config: Config) extends Acto
     }
   }
 
-  override def receive = {
+  override def receive: Receive = {
+    case DnsProtocol.Resolve(_, Srv) =>
+      sender() ! Status.Failure(
+        new IllegalArgumentException(
+          "SRV request sent to InetResolver. SRV requests are only supported by async-dns resolver"))
+
+    case r @ DnsProtocol.Resolve(name, ip @ Ip(ipv4, ipv6)) =>
+      val answer = cache.cached(r) match {
+        case Some(a) => a
+        case None =>
+          try {
+            val addresses: Array[InetAddress] = InetAddress.getAllByName(name)
```

Comment:
potentially blocking, are we on the right dispatcher for that?

Reply:
Yup, 

```
      "/IO-DNS/inet-address/*" {
        dispatcher = "akka.actor.default-blocking-io-dispatcher"
      }
```